numpy/_core/src/multiarray/ctors.c

```diff
@@ -2577,10 +2590,12 @@ PyArray_FromArrayAttr_int(
         Py_DECREF(new);
         return NULL;
     }
-    if (must_copy_but_copy_kwarg_unimplemented) {
-        /* TODO: As of NumPy 2.0 this path is only reachable by C-API. */
-        Py_SETREF(new, PyArray_NewCopy((PyArrayObject *)new, NPY_KEEPORDER));
+    if (was_copied_by__array__ != NULL && copy == 1 &&
+        must_copy_but_copy_kwarg_unimplemented == 0) {
```

Comment:
Could add a comment to remove the `was_copied_by__array__` argument again here, but it is probably clear enough (when it happens).

Reply:
Done!